File: js/tasks/fibonacci.js
/*
    Task: Fibonacci
*/
let _ = require('lodash'),
    format = require('string-format'),
    path = require('path');

function usage() {
    console.log(format('Calculate the Nth Fibonacci number\nUsage: {} <n>', path.basename(process.argv[1])));
}

function fibonacci(n) {
    f = [0, 1];
    for (let i = 2; i <= n; i++) {
        f.push(f[i-2] + f[i-1]);
    }
    //console.log(f);
    return f[n];
}

// main
let SEQ = [ 0, 1 ];
if (process.argv.length === 2) {
    usage();
    process.exit(1);
} else if (parseInt(process.argv[2]) <= 1) {
    console.log(format('F({}) = {}', process.argv[2], process.argv[2]));
    if (parseInt(process.argv[2]) < 1)
        SEQ = SEQ.splice(0, 1);
} else {
    let NUM = parseInt(process.argv[2]);
    if (NUM > 1) {
        _.range(2, NUM+1).forEach((i) => {
            SEQ.push(SEQ[i-1] + SEQ[i-2]);
        });
    }
    SIZE = SEQ.length - 1;
    console.log(format('F({}) = {}', NUM, SEQ[SIZE]));
}
console.log('Sequence:', SEQ);

File: python/tasks/fibonacci.py
'''
    Task: Fibonacci
'''
import os
import sys

def usage():
    print 'Calculate the Nth Fibonacci number\nUsage: %s <n>' % os.path.basename(sys.argv[0])

SEQ = [ 0, 1 ]
if len(sys.argv) == 1:
    usage()
    sys.exit(1)
elif int(sys.argv[1]) <= 1:
    print 'F(%s) = %s' % (sys.argv[1], sys.argv[1])
    if int(sys.argv[1]) < 1:
        del SEQ[1]
else:
    NUM = int(sys.argv[1])
    # cannot use list comprehension because it needs to index dynamically
    #SEQ.extend([ SEQ[i - 1] + SEQ[i - 2] for i in range(0, NUM) if NUM > 1 ])
    if NUM > 1:
        for i in range(2, NUM+1):
            SEQ.append(SEQ[i-1] + SEQ[i-2])
    SIZE = len(SEQ) - 1
    print 'F(%d) = %d' % (NUM, SEQ[SIZE])
print 'Sequence:', SEQ
